### PR TITLE
Functional Tests -  Add prefix verification for delivery slip file

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add prefix verification for delivery slip file
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16960)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+// Using chai
+const {expect} = require('chai');
+const helper = require('@utils/helpers');
+const loginCommon = require('@commonTests/loginBO');
+// Importing pages
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const DeliverySlipsPage = require('@pages/BO/orders/deliverySlips/index');
+const OrdersPage = require('@pages/BO/orders/index');
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js]: FYI no need for index, you can call it like this
```js
const OrdersPage = require('@pages/BO/orders');
```

File: tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    });
+  });
+
+  describe(`Back to the default delivery slip prefix value '${defaultPrefix}'`, async () => {
+    it('should go to delivery slips page', async function () {
+      await this.pageObjects.boBasePage.goToSubMenu(
+        this.pageObjects.boBasePage.ordersParentLink,
+        this.pageObjects.boBasePage.deliverySlipslink,
+      );
+      await this.pageObjects.boBasePage.closeSfToolBar();
```
Comment (boubkerbribri) [inline tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js:110]: Can you delete this please, no need to do it two times.

**Review by boubkerbribri — COMMENTED** [review]
thank you @nesrineabdmouleh

File: tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+Check the delivery slip file name
+Back to the default delivery slip prefix value
+ */
+describe('Edit delivery slip prefix and check the generated file name', async () => {
+  // before and after functions
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    this.pageObjects = await init();
+  });
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js:47]:
```suggestion
  });
  
```

File: tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  // Login into BO
+  loginCommon.loginBO();
+
+  describe(`Change the delivery slip prefix to '${deliverySlipData.prefix}'`, async () => {
+    it('should go to delivery slip page', async function () {
+      await this.pageObjects.boBasePage.goToSubMenu(
+        this.pageObjects.boBasePage.ordersParentLink,
+        this.pageObjects.boBasePage.deliverySlipslink,
+      );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js:61]:
```suggestion
      );
      
```

File: tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+      await expect(textMessage).to.contains(this.pageObjects.deliverySlipsPage.successfulUpdateMessage);
+    });
+  });
+
+  describe(`Change the order status to '${Statuses.shipped.status}' and check the file Name`, async () => {
+    it('should go to the orders page', async function () {
+      await this.pageObjects.boBasePage.goToSubMenu(
+        this.pageObjects.boBasePage.ordersParentLink,
+        this.pageObjects.boBasePage.ordersLink,
+      );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js:80]:
```suggestion
      );
      
```

File: tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      expect(fileName).to.contains(deliverySlipData.prefix.replace('#', '').trim());
+    });
+  });
+
+  describe(`Back to the default delivery slip prefix value '${defaultPrefix}'`, async () => {
+    it('should go to delivery slips page', async function () {
+      await this.pageObjects.boBasePage.goToSubMenu(
+        this.pageObjects.boBasePage.ordersParentLink,
+        this.pageObjects.boBasePage.deliverySlipslink,
+      );
```
Comment (NeOMakinG) [inline tests/puppeteer/campaigns/functional/BO/orders/deliverySlips/02_deliverySlipOptions/01_deliverySlipPrefix.js:108]:
```suggestion
      );
      
```

**Review by NeOMakinG — CHANGES_REQUESTED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**